src/index.ts

```diff
@@ -117,37 +150,54 @@ const initialize = (apiKey: string, target: Target, options: Options): Result =>
           logError(error)
         })
         .finally(() => {
-          metricsSchedulerId = setTimeout(scheduleSendingMetrics, METRICS_FLUSH_INTERVAL)
+          metricsSchedulerId = window.setTimeout(scheduleSendingMetrics, METRICS_FLUSH_INTERVAL)
         })
     } else {
-      metricsSchedulerId = setTimeout(scheduleSendingMetrics, METRICS_FLUSH_INTERVAL)
+      metricsSchedulerId = window.setTimeout(scheduleSendingMetrics, METRICS_FLUSH_INTERVAL)
     }
   }
 
+  let evaluations: Record<string, Evaluation> = {}
+
   const creatStorage = function () {
     return hasProxy
       ? new Proxy(
           {},
           {
-            get(target, featureIdentifier) {
-              if (target.hasOwnProperty(featureIdentifier)) {
-                const featureValue = target[featureIdentifier]
+            get(_storage, property) {
+              const _value = _storage[property]
+
+              if (_storage.hasOwnProperty(property)) {
+                const featureValue = _storage[property]
+                // TODO/BUG: This logic to collect metrics will fail when two variations have the same value
+                // Need to find a better way
                 const entry = metrics.find(
-                  _entry => _entry.featureIdentifier === featureIdentifier && _entry.featureValue === featureValue
+                  _entry => _entry.featureIdentifier === property && featureValue === _entry.featureValue
                 )
 
                 if (entry) {
-                  entry.count++
+                  entry.variationIdentifier = evaluations[property as string]?.identifier || ''
+                  updateMetrics(entry)
                 } else {
                   metrics.push({
-                    featureIdentifier: featureIdentifier as string,
+                    featureIdentifier: property as string,
```

Comment:
Instead of using the `as string` syntax, could you just set the type of `property` on line 167?